internal/core/services/rooms/status.go

```diff
@@ -34,26 +35,50 @@ type StatusCalculator interface {
 	CalculateRoomStatus(room game_room.GameRoom, instance game_room.Instance) (game_room.GameRoomStatus, error)
 }
 
-func NewStatusCalculator(scheduler entities.Scheduler, logger *zap.Logger) StatusCalculator {
+func NewStatusCalculator(scheduler entities.Scheduler, config RoomManagerConfig, logger *zap.Logger) StatusCalculator {
 	if scheduler.MatchAllocation != nil && scheduler.MatchAllocation.MaxMatches > 1 {
 		return &MultipleMatchStatusCalculator{
 			scheduler: scheduler,
+			config:    config,
 			logger:    logger,
 		}
 	}
 
 	return &SingleMatchStatusCalculator{
 		scheduler: scheduler,
+		config:    config,
 		logger:    logger,
 	}
 }
 
 type SingleMatchStatusCalculator struct {
 	scheduler entities.Scheduler
+	config    RoomManagerConfig
 	logger    *zap.Logger
 }
 
 func (sm *SingleMatchStatusCalculator) CalculateRoomStatus(room game_room.GameRoom, instance game_room.Instance) (game_room.GameRoomStatus, error) {
+	// Check if room is allocated and within TTL
+	if room.Status == game_room.GameStatusAllocated && room.AllocatedAt != nil {
+		timeSinceAllocation := time.Since(*room.AllocatedAt)
+		if timeSinceAllocation <= sm.config.RoomAllocationTTL {
+			// Room is still within allocation TTL
+			composedStatus, err := room.RoomComposedStatus(instance.Status.Type)
+			if err != nil {
+				return room.Status, fmt.Errorf("error calculating room composed status: %w", err)
+			}
+
+			// Allow transition to Occupied if room reports occupied
+			if composedStatus == game_room.GameStatusOccupied {
+				return game_room.GameStatusOccupied, nil
+			}
+
+			// Ignore Ready pings - keep room as Allocated
+			return game_room.GameStatusAllocated, nil
```

Comment:
What if the room pings an error state? Should we ignore or do we depend on this status to perform an action (like killing) the room in Error state?